plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenshiftUIModel.java

```diff
@@ -27,16 +27,25 @@
 import com.openshift.restclient.model.IResource;
 
 public class OpenshiftUIModel extends AbstractOpenshiftUIElement<ConnectionsRegistry, OpenshiftUIModel> {
+	
+	private static OpenshiftUIModel instance;
+	
 	private Map<IOpenShiftConnection, ConnectionWrapper> connections = new HashMap<>();
 	private List<IElementListener> listeners = new ArrayList<IElementListener>();
 
 	private IConnectionsRegistryListener listener;
 	
-	public OpenshiftUIModel() {
-		this(ConnectionsRegistrySingleton.getInstance());
+	public static OpenshiftUIModel getInstance() {

```

Comment:
unless you instantiate the instance L31, you'll have to synchronize that method to be thread safe. 
I quite like the [Initialization On Demand Holder pattern](http://blog.crazybob.org/2007/01/lazy-loading-singletons.html) as an alternative